core/tests/test_utils.py

```diff
@@ -3030,12 +3033,35 @@ def get_response_without_checking_for_errors(
                 params, dict,
                 msg='Expected params to be a dict, received %s' % params)
 
+            if http_method != 'GET':
+                self.assertIn(
+                    http_method, ['POST', 'PUT', 'DELETE'])
+
         # This swap is required to ensure that the templates are fetched from
         # source directory instead of webpack_bundles since webpack_bundles is
         # only produced after webpack compilation which is not performed during
         # backend tests.
         with self.swap(base, 'load_template', mock_load_template):
-            response = self.testapp.get(url, params=params, expect_errors=True)
+
+            if http_method == 'GET':
+                response = self.testapp.get(
+                    url, params=params, expect_errors=True
+                )
+
+            elif http_method == 'POST':
```

Comment:
This seems a little odd, why would we ever call load_template for POST,PUT, DELETE requests? I thought templates were just the html pages.

Reply:
Hey Sean, I tried putting the `with self.swap()` line inside the `if http_method == 'GET':` block like:

![image](https://github.com/oppia/oppia/assets/120438312/0abbe1bd-3964-438c-bf6b-a52a7e71c748)

So that it is only used for GET requests.
But I got an exception:

![image](https://github.com/oppia/oppia/assets/120438312/f5153116-2075-4e6f-9559-d3cde9102503)

which I tried resolving by defining `response` outside the `if`. It seems the code after `with self.swap()` does not execute if it is kept inside an `if` block. 

So I kept the swap out of the if block.

Reply:
Kindly ignore the previous comment. Successfully implemented your suggestion so that load_template is only called for GET.